frontend/src/components/Switch.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from "react";
+import { useColorMode } from "../providers/ColormodeProvider";
+
+function Switch() {
+  const [toggle, setToggle] = useState(true);
+  const toggleClass = " transform translate-x-5";
+  const { toggleTheme } = useColorMode();
+
+  const changeMode = async (mode: boolean) => {
+    setToggle(mode);
+    toggleTheme();
+  };
```

Comment:
`useColorMode` should be called in the parent component so that the `<Switch>` component can be reusable.

additionally, having 2 sets of states (ie. toggle on line 5 and the theme state inside `useColorMode`) leads to additional state sync work required, better to simplify to using only one state to avoid the chance of inconsistent behavior cause by state out of sync.

```suggestion
  const toggleClass = " transform translate-x-5";
  const { toggleTheme, theme } = useColorMode();
```